### PR TITLE
BFD-1624 Enhance bfd-db-migrator ansible role

### PR DESCRIPTION
**Pull request by mjburling**
**JIRA Ticket:**
[BFD-1624](https://jira.cms.gov/browse/BFD-1624)

**User Story or Bug Summary:**
Enhance bfd-db-migrator ansible role

---

### What Does This PR Do?
- accommodates the bfd-db-migrator's new requirements for defined and non-empty `DATABASE_USERNAME` ,`DATABASE_PASSWORD` in our GitHub Actions-based ansible Tests
- Provides templated variables for newrelic environment variables
- extends the `run-tests.sh` script to allow users to keep containers running for local debugging purposes
- tamps down on ansible warnings using the more modern/efficient `pkg` array syntax rather than the deprecated looping `with_items` array syntax.

### What Should Reviewers Watch For?

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: ops/ansible/roles/bfd-db-migrator/templates/bfd-db-migrator-service.sh.j2 [hunk 1]
```diff
@@ -2,6 +2,7 @@
 export DATABASE_URL='{{ db_migrator_db_url }}'
 export DATABASE_USERNAME='{{ db_migrator_db_username }}'
 export DATABASE_PASSWORD='{{ db_migrator_db_password }}'
+export DATABASE_MAX_POOL_SIZE='{{ db_migrator_db_max_pool_size }}'
```
Comment (christrotteradhoc) [inline ops/ansible/roles/bfd-db-migrator/templates/bfd-db-migrator-service.sh.j2:5]: You mentioned that this somehow became required by the java app which seems wrong to me? It seems weird to have to set this. The java code has the reasonable default of 1 already. Maybe there is a bug in the migrator app handling?

Comment (christrotteradhoc) [reply ops/ansible/roles/bfd-db-migrator/templates/bfd-db-migrator-service.sh.j2:5]: this is a bug. I left on comment on the other PR just now.

File: ops/ansible/roles/bfd-db-migrator/test/run-tests.sh [hunk 2]
```diff
@@ -1,22 +1,51 @@
...
 # Stop immediately if any command returns a non-zero result.
 set -eou pipefail
 
-# These variables can be adjusted to change which test is run.
+function help() {
+    echo "This script runs our test cases locally, via Docker."
+    echo "Usage: ${0} [-hk] [image id]"
+    echo "Options:"
+    echo "  ${0} -h:      [h]elp displays this message and exits."
+    # TODO: complete the getopts implementation
```
Comment (lsmitchell) [inline ops/ansible/roles/bfd-db-migrator/test/run-tests.sh:11]: Please add a ticket to this TODO to track this future work

**Review by lsmitchell — COMMENTED** [review]
Approved barring one thing!

**Review by lsmitchell — APPROVED** [review]

**Review by lsmitchell — APPROVED** [review]

**Review by christrotteradhoc — APPROVED** [review]

**Pull request merged**